src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -412,24 +413,22 @@ protected function instantiateObject(array &$data, string $class, array &$contex
                         true
                     );
                     $context['not_normalizable_value_exceptions'][] = $exception;
-
-                    return $reflectionClass->newInstanceWithoutConstructor();
                 }
             }
 
-            if ($constructor->isConstructor()) {
-                try {
-                    return $reflectionClass->newInstanceArgs($params);
-                } catch (\TypeError $th) {
-                    if (!isset($context['not_normalizable_value_exceptions'])) {
-                        throw $th;
-                    }
```

Comment:
This was originally added in #50024, but I think it's not necessary any more because of the new `if` on line 419.